corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py

```diff
@@ -31,6 +33,67 @@
 SHARED_ID = "our_group"
 PARENT_TYPE = "mother"
 
+
+def NOOP(*args, **kwargs):
+    pass
+
+
+class RunConfig(object):
+
+    def __init__(self, settings, pre_run=None, post_run=None):
+        self.settings = settings
+        self.pre_run = pre_run or NOOP
+        self.post_run = post_run or NOOP
+
+
+class RunWithMultipleConfigs(object):
```

Comment:
stole a lot of this from `RunWithMultipleSettings` but needed to extend it a bit. wasn't sure it was helpful to push this into there so just copied into a new class. cc @dannyroberts